apps/constellation/settings.cpp

```diff
@@ -43,6 +43,44 @@ const uint32_t DEFAULT_MAX_PEERS          = 3;
 const uint32_t DEFAULT_TRANSIENT_PEERS    = 1;
 const uint32_t NUM_SYSTEM_THREADS = static_cast<uint32_t>(std::thread::hardware_concurrency());
 
+class ArgvWrapper
+{
+public:
+  explicit ArgvWrapper(Settings::UpdateBatch const &batch)
+    : batch_(batch)
+    , argc_((batch_.size() * 2) + 1)
+    , argv_(argc_, nullptr)
+  {
+    argv_[0] = "wrapper";
+
+    std::size_t idx{1};
+    for (auto const &element : batch_)
+    {
+      assert(idx + 1 < argv_.size());
+
+      argv_[idx]     = element.first.c_str();
+      argv_[idx + 1] = element.second.c_str();
+
+      idx += 2;
+    }
+  }
+
+  int argc() const
+  {
+    return static_cast<int>(argc_);
+  }
+
+  char **argv() const
+  {
+    return const_cast<char **>(argv_.data());
```

Comment:
!

Just make `UpdateFromArgs` accept `char const **`. There's some knock-on effects from that, but it's better than casting away `const`. I know there is pressure for a new release, but this is a risk we can do without.

Also, you seem to be at least partly rewriting `SystemArgAdapter` from `setting_collection_tests.cpp`.

`UpdateBatch` does not seem to exist on `master`. Is this intentional?